Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php

```diff
@@ -122,10 +122,13 @@ protected function initializeAction()
      */
     public function indexAction()
     {
-        $currentAccount = $this->securityContext->getAccount();
-        $userWorkspace = $this->workspaceRepository->findOneByName(UserUtility::getPersonalWorkspaceNameForUsername($currentAccount->getAccountIdentifier()));
-        /** @var Workspace $userWorkspace */
+        $userWorkspace = $this->workspaceRepository->getPersonalWorkspaceByOwnerIdentifier(
+            $this->persistenceManager->getIdentifierByObject(
```

Comment:
Maybe it makes sense to move the ID resolution to the `UserService` so we could write sth like `-> getPersonalWorkspaceByOwnerIdentifier($this->userService->getCurrentUserIdentifier())` in those places?

Reply:
or move it to the repository maybe? Then we could have `->getPersonWorkspaceByOwner($user);`

Reply:
In general I'd say we should pass around the object only where it is absolutely required. Otherwise we'll have to fetch the object first in places where we only have the ID